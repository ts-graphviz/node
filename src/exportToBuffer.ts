import { IRootCluster, toDot } from 'ts-graphviz';
import { ExecuteOption } from './types';
import { executeDot } from './executeDot';

type Option = Omit<ExecuteOption, 'output'>;

/**
 * Returns the Graphviz output result as a buffer.
 */
export async function exportToBuffer(dot: IRootCluster | string, options: Option = {}): Promise<Buffer> {
  const input = typeof dot === 'string' ? dot : toDot(dot);
  return await executeDot(input, options);
}
