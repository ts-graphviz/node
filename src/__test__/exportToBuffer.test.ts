jest.mock('../utils');
jest.mock('../executeDot');
jest.mock('tmp-promise');

import { digraph } from 'ts-graphviz';
import { exportToBuffer } from '../exportToBuffer';

describe('exportToBuffer', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test('input can be given as string type', async () => {
    const dot = 'digraph g { a -> b; }';
    const result = await exportToBuffer(dot);
    expect(result.toString()).toMatchSnapshot();
  });

  test('input can be given as RootCluster type', async () => {
    const dot = digraph('g', (g) => {
      g.edge(['a', 'b']);
    });
    const result = await exportToBuffer(dot);
    expect(result.toString()).toMatchSnapshot();
  });
});
