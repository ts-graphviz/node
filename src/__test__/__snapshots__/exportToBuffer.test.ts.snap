// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exportToBuffer input can be given as RootCluster type 1`] = `
"digraph \\"g\\" {
  \\"a\\" -> \\"b\\";
}"
`;

exports[`exportToBuffer input can be given as string type 1`] = `"digraph g { a -> b; }"`;
